include/sys/zil.h

```diff
@@ -474,10 +475,39 @@ typedef struct zil_stats {
 	kstat_named_t zil_itx_metaslab_slog_bytes;
 } zil_stats_t;
 
-#define	ZIL_STAT_INCR(stat, val) \
-    atomic_add_64(&zil_stats.stat.value.ui64, (val));
-#define	ZIL_STAT_BUMP(stat) \
-    ZIL_STAT_INCR(stat, 1);
+typedef struct zil_sums {
+	wmsum_t zil_commit_count;
+	wmsum_t zil_commit_writer_count;
+	wmsum_t zil_itx_count;
+	wmsum_t zil_itx_indirect_count;
+	wmsum_t zil_itx_indirect_bytes;
+	wmsum_t zil_itx_copied_count;
+	wmsum_t zil_itx_copied_bytes;
+	wmsum_t zil_itx_needcopy_count;
+	wmsum_t zil_itx_needcopy_bytes;
+	wmsum_t zil_itx_metaslab_normal_count;
+	wmsum_t zil_itx_metaslab_normal_bytes;
+	wmsum_t zil_itx_metaslab_slog_count;
+	wmsum_t zil_itx_metaslab_slog_bytes;
+} zil_sums_t;
+
+#ifdef _KERNEL
+#define	ZIL_STAT_INCR(zil, stat, val) \
+	do { \
+		int64_t tmpval = (val); \
+		wmsum_add(&zil_sums_global.stat, tmpval); \
+		wmsum_add(&zil->zil_sums->stat, tmpval); \
+	} while (0)
+#else
+#define	ZIL_STAT_INCR(zil, stat, val) \
```

Comment:
This one can be simplified to just the one line that does anything.

Reply:
As per our discussion on slack, added a NULL check.